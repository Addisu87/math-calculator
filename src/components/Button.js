import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => <input type="button" value={props.label} />;

Button.propTypes = {
  label: PropTypes.string.isRequired
};

export default Button;
