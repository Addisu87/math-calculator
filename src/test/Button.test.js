import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Button from '../components/Button';

describe('Navigation bar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Button />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
