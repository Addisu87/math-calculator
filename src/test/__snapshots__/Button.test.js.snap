// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar renders correctly 1`] = `
<input
  className="btn-undefined"
  type="button"
/>
`;
